landscape/lib/sysstats.py

```diff
@@ -104,17 +108,27 @@ class ThermalZone(object):
     def __init__(self, base_path, name):
         self.name = name
         self.path = os.path.join(base_path, name)
-        temperature_path = os.path.join(self.path, "temperature")
+        temperature_path = os.path.join(self.path, "temp")
         if os.path.isfile(temperature_path):
-            for line in open(temperature_path):
-                if line.startswith("temperature:"):
-                    self.temperature = line[12:].strip()
-                    try:
-                        value, unit = self.temperature.split()
-                        self.temperature_value = int(value)
-                        self.temperature_unit = unit
-                    except ValueError:
-                        pass
+            with open(temperature_path) as f:
+                line = f.readline()
+                if re.match(r'^\d+\s*$', line):
```

Comment:
I'd rather use a `try: except` block around `int()` as in the `else:` codepath as the most common, non-breaking case should be much faster.

Reply:
Good catch, fixed.